storage/badger/dkg_state.go

```diff
@@ -219,21 +224,74 @@ func (ds *RecoverablePrivateBeaconKeyStateMachine) RetrieveMyBeaconPrivateKey(ep
 	return
 }
 
+// CommitMyBeaconPrivateKey commits the previously inserted random beacon private key for an epoch.
+// Effectively, this method transitions the state machine into the [flow.RandomBeaconKeyCommitted] state if the current state is [flow.DKGStateCompleted].
+// Caller needs to supply the [flow.EpochCommit] which is an evidence that the key has been indeed included for the given epoch.
+// No errors are expected during normal operations.
+func (ds *RecoverablePrivateBeaconKeyStateMachine) CommitMyBeaconPrivateKey(epochCounter uint64, commit *flow.EpochCommit) error {
+	return operation.RetryOnConflictTx(ds.db, transaction.Update, func(tx *transaction.Tx) error {
+		currentState, err := retrieveCurrentStateTx(epochCounter)(tx.DBTxn)
+		if err != nil {
+			return err
+		}
+		// if we are in committed state then there is nothing to do
+		if currentState == flow.RandomBeaconKeyCommitted {
+			return nil
+		}
+		key, err := ds.keyCache.Get(epochCounter)(tx.DBTxn)
+		if err != nil {
+			return storage.NewInvalidDKGStateTransitionErrorf(currentState, flow.RandomBeaconKeyCommitted, "cannot transition without a valid random beacon key: %w", err)
+		}
+
+		// verify that the key is part of the EpochCommit
+		if err = ensureKeyIncludedInEpoch(epochCounter, key, commit); err != nil {
+			return storage.NewInvalidDKGStateTransitionErrorf(currentState, flow.RandomBeaconKeyCommitted,
+				"previously storred key has not been found in epoch commit event: %w", err)
+		}
+		return ds.processStateTransition(epochCounter, currentState, flow.RandomBeaconKeyCommitted)(tx)
+	})
+}
+
 // UpsertMyBeaconPrivateKey overwrites the random beacon private key for the epoch that recovers the protocol
 // from Epoch Fallback Mode. State transitions are allowed if and only if the current state is not equal to
 // [flow.RandomBeaconKeyCommitted]. The resulting state of this method call is [flow.RandomBeaconKeyCommitted].
```

Comment:
```suggestion
	// from Epoch Fallback Mode. The resulting state of this method call is [flow.RandomBeaconKeyCommitted].
	// State transitions are allowed if and only if the current state is not equal to [flow.RandomBeaconKeyCommitted]. 
	// Repeated calls for the same epoch are idempotent, if and only if the provided EpochCommit confirms the already
	// committed key (error otherwise).
```